src/Dfc.CourseDirectory.WebV2/ViewHelpers/ProviderContextHelper.cs

```diff
@@ -1,19 +1,27 @@
 ﻿using System;
-using Dfc.CourseDirectory.WebV2.HttpContextFeatures;
+using System.Collections.Generic;
+using Dfc.CourseDirectory.WebV2.Filters;
 using Flurl;
-using Microsoft.AspNetCore.Mvc.Rendering;
-using Microsoft.AspNetCore.Mvc.ViewFeatures;
 
 namespace Dfc.CourseDirectory.WebV2.ViewHelpers
 {
-    public class ProviderContextHelper : IViewContextAware
+    public class ProviderContextHelper
     {
-        private ViewContext _viewContext;
+        private readonly IProviderContextProvider _providerContextProvider;
+
+        public ProviderContextHelper(IProviderContextProvider providerContextProvider)
+        {
+            _providerContextProvider = providerContextProvider;
+        }
 
         public ProviderInfo ProviderInfo => ProviderContext?.ProviderInfo;
 
-        private ProviderContext ProviderContext =>
-            _viewContext.HttpContext.Features.Get<ProviderContextFeature>()?.ProviderContext;
+        public IDictionary<string, string> RouteValues => new Dictionary<string, string>()
```

Comment:
`IReadOnlyDictionary<string, string>`? :)

Reply:
Has to be the mutable one here sadly so it can be passed to tag helper without casting